src/test/java/it/gov/pagopa/payhub/activities/activity/SendEmailIngestionFlowActivityTest.java

```diff
@@ -0,0 +1,194 @@
+import it.gov.pagopa.payhub.activities.config.EmailTemplatesConfiguration;
+import it.gov.pagopa.payhub.activities.dao.IngestionFlowFileDao;
+import it.gov.pagopa.payhub.activities.dto.*;
+import it.gov.pagopa.payhub.activities.service.SendMailService;
+import it.gov.pagopa.payhub.activities.service.UserAuthorizationService;
+import jakarta.mail.MessagingException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.TestPropertySource;
+
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+
+/**
+ * test class for send mail for ingestion activity
+ */
+@SpringBootTest(
+  classes = {
+    SendEmailIngestionFlowActivityImpl.class,
+    EmailTemplatesConfiguration.class
+  },
+  webEnvironment = SpringBootTest.WebEnvironment.NONE)
+@TestPropertySource(properties = {
+    "activity.root.path=/DATA/"
+})
+@EnableConfigurationProperties
+@ExtendWith(MockitoExtension.class)
+class SendEmailIngestionFlowActivityTest {
+    @Autowired
+    private SendEmailIngestionFlowActivity sendEmailIngestionFlowActivity;
+    @MockBean
+    private IngestionFlowFileDao ingestionFlowFileDao;
+    @MockBean
+    private SendMailService sendMailService;
+    @MockBean
+    private UserAuthorizationService userAuthorizationService;
+
+    private IngestionFlowFileDTO validIngestionFlowFileDTO;
+    private IngestionFlowFileDTO invalidIngestionFlowFileDTO;
+    private UserInfoDTO validUserInfoDTO;
+    private UserInfoDTO invalidUserInfoDTO;
+    private UserInfoDTO validOrganizationInfoDTO;
+
+    @BeforeEach
+    void init() {
+        createBeans();
+    }
+
+    /**
+     * send email for OK loading flow, success: valid user and valid flow
+     */
+    @Test
+    void sendEmailIngestionOkValidUserValidFlowSuccess() throws MessagingException {
+        Long ingestionFlowFileId = 100L;
+        Mockito.when(ingestionFlowFileDao.findById(ingestionFlowFileId)).thenReturn(Optional.of(validIngestionFlowFileDTO));
+        Mockito.when(userAuthorizationService.getUserInfo(validIngestionFlowFileDTO.getOrg().getIpaCode(), validUserInfoDTO.getMappedExternalUserId())).thenReturn(validUserInfoDTO);
+        Mockito.when(userAuthorizationService.getUserInfo(validIngestionFlowFileDTO.getOrg().getIpaCode(),validIngestionFlowFileDTO.getOrg().getIpaCode()+"-WS_USER")).thenReturn(validOrganizationInfoDTO);
+        Mockito.doNothing().when(sendMailService).sendMail(any(MailTo.class));
```

Comment:
Instead of any MailTo, could we set an expectedMailTo?
This way we will test its right building logic

Reply:
OK, done